### ISSUE TITLE
Proactive Flushes: use env max flushes as flush limit

### ISSUE THREAD

**Issue by Yuval-Ariel** [speedb-io/speedb#360]
i dont see a reason why a user should set max simultaneous flushes for the proactive WBM while theres already a setting for the max bg flushes a group of dbs can use. 

seems logical to me that this configuration can be spared from the user and just take the number of max parallel flushes from the env (how many high priority threads there are).

**Comment by Yuval-Ariel** [comment]
what do you think @udi-speedb ?

**Comment by Yuval-Ariel** [comment]
1. options->env->GetBackgroundThreads(Priority pri) (e.g in env/env_posix.cc)
2. its per Env and the env applies to all dbs which have been given that env. so all the dbs using the same env will have a total limit of High priority threads.
3. yes its possible through SetBackgroundThreads
4. im not sure it makes sense not to use the same env for all dbs using the WBM. need to research. @hilikspdb ?
5. the number of concurrent threads that should be run on all dbs is bound by the thread pool in any case so i dont see any harm in this.
6. im not suggesting we should start working on it now. just opened it to discuss

**Comment by mrambacher** [comment]
Many of those options are part of the DBOptions (and not properties of the Env).  This means that the "last DBOptions" in would win on the settings, if there are multiple DBs.

**Comment by hilikspdb** [comment]
At least in the case of "background flush" the db_option would only set a
higher value.  Regardless, we need a multiple database configuration file
that will collect all the objects that may be common to multiple databases
(cache , wbm, background jobs etc) and add to them (for example we need a
log file or else statistics of usage etc) ...

**Comment by udi-speedb** [comment]
@Yuval-Ariel - I'll just refer to #5 and creating new dependencies. In general, an infrastructure entity such as the WBM, which is currently independent of most other entities, is a very good thing and should be kept as such. 
It may be possible to support your proposal in other means, without creating dependencies, however, I will leave that to when we actually decide to do anything about it.

**Comment by Yuval-Ariel** [comment]
what happens when a user sets WBM.max_flushes = 4, but the env thread pool for these dbs only supports max 3 flush threads?

**Comment by udi-speedb** [comment]
The WBM will **try** to initiate 4.
As for what happens afterwards, I can't say for sure. I can only guestimate: I assume that a flush job would be created and wait is turn to run.
@hilikspdb - Could you please answer?

**Comment by hilikspdb** [comment]
Additional flushes will wait in the flush queue and be executed once a
flush will end

**Comment by Guyme** [comment]
@hilikspdb @udi-speedb @Yuval-Ariel @mrambacher - Final thoughts ?

**Comment by hilikspdb** [comment]
The relationships between database options and multi-database options are
too complex and need work to simplify them. This should not impact the
current project

**Comment by Guyme** [comment]
@Yuval-Ariel - Are you satisfied with this answer or you still think we can use the same paramater ?

**Comment by Yuval-Ariel** [comment]
i agree this should be done as part of a bigger project of correctly setting and sanitizing options for multi-db setup

**Comment by Yuval-Ariel** [comment]
and perhaps a note in the documentation of this feature explaining to the user how to set the WBM.max_flushes and its connection to the env's max flush threads.
